### DISCUSSION TITLE
HTMX and _hyperscript breaks if I go "back" using browser back button

### DISCUSSION THREAD

**Discussion by yaroslav-gwit** [unpoly/unpoly#422] [Q&A] [answered]
Hi guys, first of all thank you for this amazing project, it's doing wonders for me. Just few words about myself, so you can understand where I am coming from: I am a system admin/devops guy, JS is very new to me, and I am trying to avoid it where possible.

I am writing a new pet project/website (Django with _hyperscript, htmx and unpoly) for Ukrainians in London. Everything is going well so far, but there is an issue: nearly every link in the navbar is a [up-target] link, which is represented like so in Django
`<a href="{% url 'MainPage' %}" _="on click remove .active from .nav-link then add .active to me then log 'link clicked'" up-target=".unpoly-content" up-transition="cross-fade" up-history='true' class="nav-link px-2{% if request.path == MainPage %} active{% endif %}">`. The `_="on click remove .active from .nav-link then add .active to me then log 'link clicked'"` portion is a _hyperscript that makes a `nav-item` active any time it's clicked. Everything works as expected, until I hit a `back` button in the browser, _hyperscript stops working and although I can navigate just fine, `nav-item` classes don't get updated anymore.

Has anyone encountered this before? If so what was your solution?

**Comment by triskweline** [comment, marked as answer]
I'm not using hyperscript, but there's an [issue with a similar problem](https://github.com/bigskysoftware/_hyperscript/issues/133) in the hyperscript repo.

Judging from the replies in the issues, this might fix it for you:

```js
up.on('up:fragment:inserted', (event, fragment) => {
  _hyperscript.processNode(fragment)
})
```

Also note that [Unpoly already adds classes to links that are loading or that are pointing to the current page](https://unpoly.com/up.feedback).

**Reply by yaroslav-gwit** [reply]
This is brilliant, thank you! After looking at he doc link you've sent, this was my fix (this will work with any Bootstrap deployment):
`up.feedback.config.currentClasses = ['active']`

I'll try the _hyperscript method later on today :)

**Reply by yaroslav-gwit** [reply]
Thanks to your answer, I was able to figure out what I had to do to integrate HTMX, _hyperscript and Unpoly:
- Create a separate JS file to re-initialise HTMX and _hyperscript on every page load done by Unpoly
```
up.on('up:fragment:inserted', () => {
  htmx.process(document)
  _hyperscript.processNode(document)
})
```
- Link the JS files in the `head` section of the page in this exact order (own.js is the file created earlier)
```
<script defer src="https://unpkg.com/unpoly@2.6.1/unpoly.min.js"></script>
<link rel="stylesheet" href="https://unpkg.com/unpoly@2.6.1/unpoly.min.css">
<script defer src="https://unpkg.com/hyperscript.org@0.9.7"></script>
<script defer src="https://unpkg.com/htmx.org@1.8.0"></script>
<script defer src="{% static 'js/own.js' %}"></script>
```
- Ctrl+Shift+R -> reload the page to apply the changes

Now each lib has it's own purpose:
- _hyperscript to make quick and dirty changes to the DOM
- HTMX to submit and verify complex forms
- Unpoly to give my Django App a `SPA feel (c)`

I am very satisfied with the result. Thank you very much for your help again.
